File: src/main/java/com/dansoftware/libraryapp/gui/info/dependency/meta/DependencyInfo.java
package com.dansoftware.libraryapp.gui.info.dependency.meta;

import java.util.Optional;

/**
 * A {@link DependencyInfo} represents a software that is used by this application.
 *
 * @author Daniel Gyorffy
 */
public class DependencyInfo {

    private final String name;
    private final String websiteUrl;
    private final LicenseInfo licenseInfo;

    public DependencyInfo(String name, String websiteUrl, LicenseInfo licenseInfo) {
        this.name = name;
        this.websiteUrl = websiteUrl;
        this.licenseInfo = licenseInfo;
    }

    public String getName() {
        return name;
    }

    public Optional<String> getWebsiteUrl() {
        return Optional.ofNullable(websiteUrl);
    }

    public LicenseInfo getLicenseInfo() {
        return licenseInfo;
    }
}

File: src/main/java/com/dansoftware/boomega/gui/info/dependency/meta/DependencyInfo.kt
package com.dansoftware.boomega.gui.info.dependency.meta

import java.util.*

/**
 * A [DependencyInfo] represents a software that is used by this application.
 *
 * @author Daniel Gyorffy
 */
data class DependencyInfo(val name: String, private val websiteUrl: String, val licenseInfo: LicenseInfo) {
    fun getWebsiteUrl(): Optional<String> = Optional.ofNullable(websiteUrl)
}